core/src/main/scala/kafka/admin/ConsumerGroupCommand.scala

```diff
@@ -395,6 +401,83 @@ object ConsumerGroupCommand extends Logging {
       result
     }
 
+    def deleteOffsets(groupId: String, topics: List[String]): Map[TopicPartition, Throwable] = {
+      var result: Map[TopicPartition, Throwable] = mutable.HashMap()
+
+      val (topicWithPartitions, topicWithoutPartitions) = topics.partition(_.contains(":"))
+
+      val knownPartitions = topicWithPartitions.flatMap { topicArg =>
+        val split = topicArg.split(":")
+        split(1).split(",").map { partition =>
+          new TopicPartition(split(0), partition.toInt)
+        }
+      }
+
+      // Get the partitions of topics that the user did not explicitly specify the partitions
+      val describeTopicsResult = adminClient.describeTopics(
+        topicWithoutPartitions.asJava,
+        withTimeoutMs(new DescribeTopicsOptions))
+
+      val unknownPartitions = describeTopicsResult.values().asScala.flatMap { case (topic, future) =>
+        Try(future.get()) match {
+          case Success(description) => description.partitions().asScala.map { partition =>
+            new TopicPartition(topic, partition.partition())
+          }
+          case Failure(e) =>
+            result += new TopicPartition(topic, -1) -> e
+            List.empty
+        }
+      }
+
+      val partitions = knownPartitions ++ unknownPartitions
+
+      val deleteResult = adminClient.deleteConsumerGroupOffsets(
+        groupId,
+        partitions.toSet.asJava,
+        withTimeoutMs(new DeleteConsumerGroupOffsetsOptions)
+      )
+
+      deleteResult.all().get()
+
+      partitions.foreach { partition =>
+        Try(deleteResult.partitionResult(partition).get()) match {
+          case Success(_) => result += partition -> null
+          case Failure(e) => result += partition -> e
+        }
+      }
+
+      result
+    }
+
+    def deleteOffsets(): Unit = {
+      val groupId = opts.options.valueOf(opts.groupOpt)
+      val topics = opts.options.valuesOf(opts.topicOpt).asScala.toList
+
+      try {
+        val result = deleteOffsets(groupId, topics)
+
+        println("\n%-30s %-15s %-15s".format("TOPIC", "PARTITION", "STATUS"))
+        result.toList.sortBy(t => t._1.topic + t._1.partition.toString).foreach { case (tp, error) =>
+          println("%-30s %-15s %-15s".format(
+            tp.topic,
+            if (tp.partition >= 0) tp.partition else "Not Provided",
+            if (error != null) s"Error: ${error.getMessage}" else "Successful"
+          ))
+        }
+      } catch {
+        case e: ExecutionException =>
+          Errors.forException(e.getCause) match {
+            case Errors.INVALID_GROUP_ID | Errors.GROUP_ID_NOT_FOUND =>
+              printError(s"'$groupId' is not valid or does not exist.")
```

Comment:
nit: why not use separate cases and make it clear which it is?

Reply:
good point. i have no good reason to be honest. i felt like they were good together as they are close. i can do a small PR to change it.